### PR TITLE
IEP-1026: Pressing "Stop" button during build process leads to error.

### PR DESCRIPTION
**Pull request by sigmaaa**
## Description

Pressing Stop now terminates building processes. Also did a small refactoring, which improves build time for DFU.

Fixes # ([IEP-1026](https://jira.espressif.com:8443/browse/IEP-1026))

## Type of change

Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How has this been tested?

Test 1:
build -> Stop
Test 2:
select DFU -> Build -> Stop

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- DFU build
- Build

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **Refactor**
  - Improved build configuration management and streamlined build process.
  - Enhanced DFU (Device Firmware Update) command utility for better performance and maintainability.

- **Documentation**
  - Updated system requirements to include a specific version dependency for enhanced compatibility.

### CODE REVIEW

File: bundles/com.espressif.idf.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -8,7 +8,7 @@ Bundle-Vendor: %Bundle-Vendor
 Require-Bundle: org.eclipse.core.runtime,
  org.eclipse.core.filesystem;visibility:=reexport,
  org.eclipse.core.resources;visibility:=reexport,
- org.eclipse.cdt.core,
+ org.eclipse.cdt.core;bundle-version="8.0.0",
```
Comment (alirana01) [inline bundles/com.espressif.idf.core/META-INF/MANIFEST.MF:11]: Do we need to have the version defined explicitly here?🤔

**Comment by AndriiFilippov** [comment]
@sigmaaa hi ! 

Tested under: 
OS - Windows 10 
ESP-IDF: v5.1.2 

1) Build -> Stop -> Build -> Flash -> Monitor 👍 
2) Build -> Stop -> Close Project -> Open Project -> Build -> Flash -> Monitor 👍
3) Build -> Stop -> Close Eclipse -> Open Eclipse -> Build -> Flash -> Monitor 👍
4) Build -> Stop -> Restart -> Build -> Flash - Monitor 👍 
5) select DFU -> Build -> Stop -> Build -> Flash - Monitor 👍 
6) Custom Build Folder -> Build -> Stop -> Build -> Flash - Monitor 👍 
7) New Project -> click "Run" -> Stop -> can't stop build process ❌

**Comment by sigmaaa** [comment]
Hi @AndriiFilippov, It was fixed with the last commit. @alirana01, @kolipakakondal PTAL at the last commit. I had to add a workaround and custom handler, which terminates the build process, which is part of the Launching job. To avoid handler conflicts, I created a context with id com.espressif.idf.ui.espLaunchScope and added a condition for activating a custom handler.

**Review by sigmaaa — COMMENTED** [review]
Self reviewed

**Comment by AndriiFilippov** [comment]
@sigmaaa hi ! 

Retested all scenarios. LGTM 👍

**Review by AndriiFilippov — APPROVED** [review]

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Pull request merged**